polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -105,19 +93,22 @@ class PolyKernel private[kernel] (
                   }
                   kernel.runCode(
                     id,
-                    findAvailableSymbols(prevCellIds, kernel),
-                    prevCellIds,
-                    cell.content.toString,
-                    oqSome,
-                    statusUpdates
-                  ).handleErrorWith {
+                    cellContext,
+                    cell.content.toString
+                  ).map {
+                    case (results, newCtx) =>
+                      results.to(oqSome.enqueue).compile
+                      newCtx.map { ctx =>
+                        runtimeContextHolder.put(id, ctx)
```

Comment:
Now that newCtx is an `IO` I'm not quite sure what to do with it - is there a nice way to do something when results terminates?